stetho-urlconnection/src/main/java/com/facebook/stetho/urlconnection/StethoURLConnectionManager.java

```diff
@@ -37,24 +31,37 @@
  */
 @NotThreadSafe
 public class StethoURLConnectionManager {
-  private static final AtomicInteger sSequenceNumberGenerator = new AtomicInteger(0);
+  private static boolean sIsStethoPresent = isStethoPresent();
+
+  private static boolean isStethoPresent() {
+    try {
+      Class.forName("com.facebook.stetho.Stetho");
+      return true;
+    } catch (ClassNotFoundException e) {
+      return false;
+    }
+  }
 
-  private final NetworkEventReporter mStethoHook = NetworkEventReporterImpl.get();
-  private final int mRequestId;
-  @Nullable private final String mFriendlyName;
+  private final Holder mHolder;
 
-  @Nullable private String mRequestIdString;
+  private static class Holder {
+    private final StethoURLConnectionManagerImpl impl;
 
-  private HttpURLConnection mConnection;
-  @Nullable private URLConnectionInspectorRequest mInspectorRequest;
+    public Holder(@Nullable String friendlyName) {
+      impl = new StethoURLConnectionManagerImpl(friendlyName);
+    }
+  }
 
   public StethoURLConnectionManager(@Nullable String friendlyName) {
-    mRequestId = sSequenceNumberGenerator.getAndIncrement();
-    mFriendlyName = friendlyName;
+    if (sIsStethoPresent) {
+      mHolder = new Holder(friendlyName);
+    } else {
+      mHolder = null;
+    }
   }
 
   public boolean isStethoEnabled() {
-    return mStethoHook.isEnabled();
+    return isStethoPresent() && mHolder.impl.isStethoActive();
```

Comment:
Here and throughout, consider checking (mHolder != null) rather than calling isStethoPresent() or even using sIsStethoPresent.  Because the implementation already has to be robust to being called when not active, the main goal of this check is to avoid NPE.  This intention would be clearer with the null check.